### PR TITLE
Adapt the printer demo to work on the 2.8'' display

### PR DESCRIPTION
**Pull request by ogoffart**
Part of #1015

### CODE REVIEW

File: examples/printerdemo_mcu/ui/common.slint [hunk 1]
```diff
@@ -241,7 +249,7 @@ export CheckBox := Rectangle {
             clicked => { checked = !checked; }
         }
 
-        Text {
+        if text != "" : Text {
```
Comment (tronical) [inline examples/printerdemo_mcu/ui/common.slint:252]: Smells like a future compiler optimisation :-) (remove_useless_text next to remove_useless_rectangles). Probably super rare though.

File: examples/printerdemo_mcu/ui/home_page.slint [hunk 2]
```diff
@@ -42,27 +44,26 @@ ActionButton := Rectangle {
...
 
-    property <length> button-spacing: 28px;
-    property <length> button-width: 127px;
-    property <length> button-height: button-width + 37px;
+    property <length> button-spacing: 8px;
+    property <length> button-width: 127px / 2;
+    property <length> button-height: button-width + 20px;
 
-    header: "Xerol 1347 hdp";
+    header: "Slint";
```
Comment (tronical) [inline examples/printerdemo_mcu/ui/home_page.slint:53]: Just an idea, since there's no "window title" :)

```suggestion
    header: "Slint Printer";
```

**Review by tronical — APPROVED** [review]

File: examples/printerdemo_mcu/ui/common.slint (hunk 1 above)
Comment (ogoffart) [reply examples/printerdemo_mcu/ui/common.slint:252]: that's not an optimization, that's to remove a gap in the layout  (so we don't have spacing for nothing)

File: examples/printerdemo_mcu/ui/home_page.slint (hunk 2 above)
Comment (ogoffart) [reply examples/printerdemo_mcu/ui/home_page.slint:53]: doesn't fit :-(

File: examples/printerdemo_mcu/ui/copy_page.slint [hunk 3]
```diff
@@ -27,64 +27,33 @@ export CopyPage := Page {
...
-            Rectangle {}
-
-            Label { text: "Paper Tray"; }
-            ComboBox {
-                value: "Special Tray";
-                choices: ["Special Tray", "Normal Tray"];
-            }
         }
-        Row {
+        /*Row {
```
[the remark this answers is not in the document]
Comment (ogoffart) [reply examples/printerdemo_mcu/ui/copy_page.slint:31]: i guess it could be removed.

File: examples/printerdemo_mcu/ui/home_page.slint (hunk 2 above)
Comment (tronical) [reply examples/printerdemo_mcu/ui/home_page.slint:53]: Bummer, ok :)

**Pull request merged**